syfertext/pipeline/average_doc_encoder.py

```diff
@@ -0,0 +1,188 @@
+    ) -> None:
+        """Initialize the object
+
+        Args:
+            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on their custom attribute values.
+            owner (optional): The worker on which the object is located.
+                The end user does not need to specify this argument. However,
+                it is always set by the `detail` method. If no owner is
+                found, the global LOCAL_WORKER is used instead. Actually,
+                we can alway use LOCAL_WORKER and remove the `owner` arugment
+                here, but this won't work for virtual workers where
+                `LOCAL_WORKER` does not change from worker to another.
+
+        """
+
+        self.excluded_tokens = excluded_tokens
+
+        # Set the owner
+        self.owner = owner if owner is not None else LOCAL_WORKER
+
+        # Initialize the `mpc` property that will hold SMPC config
+        # This configuration will include two workers to use to
+        # send shares and one crypto provider
+
+        # Initialize to None and only populate it if MPC is possible
+        self.mpc = None
+
+    def __call__(
+        self, doc: Union[Doc, DocPointer], crypto_config: Dict[str, object] = None
+    ) -> Union[sy.AdditiveSharingTensor, AbstractTensor]:
+        """Perform encoding of the document.
+
+        Encoding here is performed by computing the average embedding of
+        the tokens included in the doc. Some tokens specified in
+        `self.excluded_tokens` are excluded.
+
+        If an encryption mode is specified using `encryption`, then the
+        average vector is encrypted before being returned.
+
+        Args:
+            doc: The Doc object to be encoded or a pointer to it.
+            crypto_config: A dictionary containing configuration for the
+                encryption scheme chosen.
+                for example, if 'mpc' is chosen, the `config` argument
+                will have the following format:
+                - config['mpc']['node_0'] for the first node to hold
+                  the shares.
+                - config['mpc']['node_1'] for the second node to hold
+                  the shares.
+                - config['mpc']['crypto_provider'] for the node that
+                  provides crypto primitives.
+                - other configurations to be supported in the future.
+
+        Returns:
+            A Syft tensor in case no encryption is used. If in 'mpc'
+                encryption mode, the tensor returned will be a
+                AdditiveSharingTensor.
+        """
+
+        # Case 1: SMPC encryption required.
+        # In this case, `doc` is a DocPointer object although that it
+        # could also be a Doc object in theory.
+        # The returned tensor is a Syft AdditiveSharingTensor object.
+        if "mpc" in crypto_config:
+
+            # Encrypt the vector
+            vector_enc = doc.get_encrypted_vector(
```

Comment:
mpc could be done using more than 3 nodes also I think, I think better config will be like {'mpc':List[nodes]} and pass all the nodes in the list here.